### ISSUE TITLE
Type mismatch in the heartbeat interval parameter

### ISSUE THREAD

**Issue by Neverlord** [actor-framework/actor-framework#1095]
Our actor system config has the heartbeat parameter as `timespan`:

```c++
    .add<timespan>("heartbeat-interval", "interval of heartbeat messages")
```

The BASP broker reads this config parameter when spinning up:

```c++
  auto heartbeat_interval = get_or(config(), "middleman.heartbeat-interval",
                                   defaults::middleman::heartbeat_interval);
```

However, it is impossible as  a user to effectively provide a valid interval. Because:

```c++
constexpr auto heartbeat_interval = size_t{0};
```

When converting the config parameter to `timespan`, we forgot to update the BASP broker. It still believes the config parameter is an integer with millisecond resolution.

On the command line, users cannot set this parameter. CAF expects a `timespan`, so the parser complains when finding an integer without time suffix (`s`, `ms`, etc.).

Setting the parameter via configuration file as an integer works. Maybe we also should consider this a bug and reject config files with type mismatches for known parameters.

**Comment by Neverlord** [comment]
I found this bug while working on #1094.

**Comment by Neverlord** [comment]
Fixed via #1096.
